dhcpv4/bsdp/bsdp_option_default_boot_image_id.go

```diff
@@ -0,0 +1,59 @@
+// +build darwin
+
+package bsdp
+
+import (
+	"fmt"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// Implements the BSDP option default boot image ID, which tells the client
+// which image is the default boot image if one is not selected.
+
+// OptDefaultBootImageID contains the selected boot image ID.
+type OptDefaultBootImageID struct {
+	ID BootImageID
+}
+
+// ParseOptDefaultBootImageID constructs an OptDefaultBootImageID struct from a sequence of
+// bytes and returns it, or an error.
+func ParseOptDefaultBootImageID(data []byte) (*OptDefaultBootImageID, error) {
+	if len(data) < 6 {
+		return nil, dhcpv4.ErrShortByteStream
+	}
+	code := dhcpv4.OptionCode(data[0])
```

Comment:
I only now realized that we are using a different contract for DHCPv4 options, compared to DHCPv6 options. In the `dhcpv6` package, the `Parse*` methods don't require option code and length, as these are checked by the caller. While this may sound unintuitive, it removes the need to check the option twice.

I'd like to have a unique interface in the future - let's go ahead with what we have now, but then let's uniform the two

Reply:
I've opened https://github.com/insomniacslk/dhcp/issues/22